lib/apiservers/engine/backends/filter/container.go

```diff
@@ -0,0 +1,230 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filter
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/docker/engine-api/types"
+
+	"github.com/vmware/vic/lib/apiservers/engine/backends/cache"
+	"github.com/vmware/vic/lib/apiservers/portlayer/models"
+)
+
+// reused from docker/docker/daemon/list.go
+type ContainerListContext struct {
+	FilterContext
+
+	// idx is the container iteration index for this context
+	Counter int
+	// ExitCode for the passed container
+	ExitCode int
+	// exitAllowed is a list of exit codes allowed to filter with
+	exitAllowed []int
```

Comment:
@hickeng if exitAllowed is a small array, it is always preferable to use iteration instead of a map. the max size need to be benchmarked to find that edge.